.github/workflows/stack.yml

```diff
@@ -19,27 +19,40 @@ on:
     - 'test/**.hs'
 
 jobs:
-  stack:
+  auto-cancel:
     if: |
       !contains(github.event.head_commit.message, '[skip ci]')
       && !contains(github.event.head_commit.message, '[ci skip]')
       && !contains(github.event.head_commit.message, '[github skip]')
       && !contains(github.event.head_commit.message, '[skip github]')
 
+    runs-on: Ubuntu-latest # Required, but it can be anything here.
+
+    steps:
+    - uses: styfle/cancel-workflow-action@0.6.0
+      with:
+        access_token: ${{ github.token }}
+
+  stack:
+    needs: auto-cancel
+
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
         os: [ubuntu-18.04]
+        stack-ver: [2.5.1]
         ghc-ver: [8.10.2, 8.8.4, 8.6.5, 8.4.4, 8.2.2, 8.0.2]
         include:
           - os: macos-latest
             ghc-ver: 8.10.2
+            stack-ver: 2.5.1
           - os: windows-latest
             ghc-ver: 8.10.1
```

Comment:
Can this be upgraded to 8.10.2.2 or still not?

Reply:
Yes. I updated the workflow just now. However, it seems that ghc 8.10.2.2 must be installed via Chocolatey (a package manager on Windows) since it is not an official release. I am not sure if stack can get the right `ghc` to install. Otherwise, `stack` would download the broken one and throw some odd error messages which are confusing to Windows users. 

I left comments in `stack-8.10.2.2.yaml`.